api/search_test.go

```diff
@@ -0,0 +1,155 @@
+		name          string
+		query         []string
+		expectedCodes []string
+	}{
+		{
+			name:          "matching single query",
+			query:         []string{"ind1"},
+			expectedCodes: []string{"IND1"},
+		},
+		{
+			name:          "matching multiple queries",
+			query:         []string{"ind1", "ind2"},
+			expectedCodes: []string{"IND1", "IND2"},
+		},
+		{
+			name:          "no matching queries",
+			query:         []string{"foo", "bar"},
+			expectedCodes: []string{},
+		},
+		// algorithm of PrefixMap is depth first search
+		// so It will get the data in reverse
+		// keep that in mind when updating tests
+		{
+			name:          "matching partial query",
+			query:         []string{"ind"},
+			expectedCodes: []string{"IND3", "IND2", "IND1"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			matchingIndustries := getAllMatchingIndustries(tt.query, mockDB)
+			assert.Equal(t, len(tt.expectedCodes), len(matchingIndustries), "expected %d matching industries, got %d", len(tt.expectedCodes), len(matchingIndustries))
+			for i, industryResp := range matchingIndustries {
+				assert.Equal(t, tt.expectedCodes[i], industryResp.Code, "expected industry with code %s, got %s", tt.expectedCodes[i], industryResp.Code)
+			}
+		})
+	}
+}
+
+func TestGetAllMatchingAreas(t *testing.T) {
+	// get a mock ScrubberDB with some areas
+	mockDB := mock.DB()
+	tests := []struct {
+		name          string
+		query         []string
+		expectedNames []*models.AreaResp
+	}{
+		{
+			name:  "matching single query",
+			query: []string{"OAC1"},
+			expectedNames: []*models.AreaResp{
+				{
+					Name:       "LAN1",
+					Region:     "RN1",
+					RegionCode: "RC1",
+					Codes: map[string]string{
+						"OAC1": "OAC1",
+					},
+				},
+			},
+		},
+		{
+			name:  "matching multiple queries",
+			query: []string{"OAC1", "OAC2"},
+			expectedNames: []*models.AreaResp{
+				{
+					Name:       "LAN1",
+					Region:     "RN1",
+					RegionCode: "RC1",
+					Codes: map[string]string{
+						"OAC1": "OAC1",
+					},
+				},
+				{
+					Name:       "LAN2",
+					Region:     "RN2",
+					RegionCode: "RC2",
+					Codes: map[string]string{
+						"OAC2": "OAC2",
+					},
+				},
+			},
+		},
+		{
+			// PrefixMap algorithm is depth first search
+			// so when running partial queries it will get
+			// the last area first, keep that in mind when updating tests
+			name:  "matching partial queries",
+			query: []string{"OAC"},
+			expectedNames: []*models.AreaResp{
+				{
+					Name:       "LAN3",
+					Region:     "RN3",
+					RegionCode: "RC3",
+					Codes: map[string]string{
+						"OAC3": "OAC3",
+					},
+				},
+				{
+					Name:       "LAN2",
+					Region:     "RN2",
+					RegionCode: "RC2",
+					Codes: map[string]string{
+						"OAC2": "OAC2",
+					},
+				},
+				{
+					Name:       "LAN1",
+					Region:     "RN1",
+					RegionCode: "RC1",
+					Codes: map[string]string{
+						"OAC1": "OAC1",
+					},
+				},
+			},
+		},
```

Comment:
I don't think this will get past your extraction regex either so is not a valid test case.